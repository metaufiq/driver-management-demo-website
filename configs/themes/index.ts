const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };