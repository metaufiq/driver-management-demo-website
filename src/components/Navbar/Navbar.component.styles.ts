import styled from 'styled-components'

import { StyledComponentProps } from './Navbar.component.types'

export const Container = styled.div<StyledComponentProps>`
    display: flex;
    justify-content: ${({visible})=>visible ? 'flex-end': 'space-between'};
    align-items: center;
    height: 3.5rem;
    padding: 0 2rem;
    background-color: white;
    @media (min-width: 768px) {
      justify-content: flex-end;
    }
`

export const InlineText = styled.p`
  display: inline-block;
`

export const UserName = styled(InlineText)`
  color: ${({theme})=>theme.colors.primary};
  margin-left: 0.3rem;
`

export const IconBurgerContainer = styled.a<StyledComponentProps>`
    display: ${({visible})=>visible ? 'none': 'flex'};
    justify-content: center;
    font-size: 1.5rem;
    color: ${({theme})=>theme.colors.primary};
    @media (min-width: 768px) {
      display: none;
    }
`

export const UserContainer = styled.div`
  align-items: center;
  justify-content: flex-end;
  display: flex;
  flex-direction: row;
`
