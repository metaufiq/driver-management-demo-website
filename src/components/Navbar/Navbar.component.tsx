import { ReactElement } from "react";
import {FaBars} from 'react-icons/fa'

import { Component } from "@dmd/types";
import { USER_AVATAR_URL } from "../../constants";
import Avatar from "../Avatar";
import { Container, IconBurgerContainer, InlineText, UserContainer, UserName } from "./Navbar.component.styles";
import { Props } from "./Navbar.component.types";

/**
 * Navbar components
 * @param {Props} props component props 
 * @returns {ReactElement} Navbar components
 */
const Navbar:Component<Props> = ({visible, onPressBurger}):ReactElement=>(
  <Container visible={visible}>
      <IconBurgerContainer
        visible={visible}
        href="#" 
        onClick={onPressBurger}
      >
        <FaBars />
      </IconBurgerContainer>
      <UserContainer>
        <InlineText>Hello,</InlineText>
        <UserName>Shipper User</UserName>
        <Avatar 
          src={USER_AVATAR_URL}
          height={'30%'}
          width={'30%'}
        />
      </UserContainer>
  </Container>
)

export default Navbar