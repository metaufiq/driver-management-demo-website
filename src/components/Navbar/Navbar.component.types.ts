export type StyledComponentProps = {
  visible: boolean,
}

export type Props = StyledComponentProps & {
  onPressBurger: VoidFunction,
  userImage: string, 
}