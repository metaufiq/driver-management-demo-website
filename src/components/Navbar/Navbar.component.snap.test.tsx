import { assertSnapshots } from "@dmd/test";
import Navbar from ".";
import { Configs } from "@dmd/test/index.types";
import { Props } from "./Navbar.component.types";
import { themes } from "../../configs/themes";

describe('NavbarSnapshot', () => {

  const configs: Configs<Props> = [
    {
      desc: 'should render normal Navbar with visible value is equal to true',
      props:{
        visible:true,
        userImage: 'user_image_url',
        onPressBurger: ()=>{}
      },
      theme: themes,
      useTheme: true
    },
    {
      desc: 'should render normal Navbar with visible value is equal to false',
      props:{
        visible:false,
        userImage: 'user_image_url',
        onPressBurger: ()=>{},
      },
      theme: themes,
      useTheme: true
    }
  ];

  assertSnapshots(Navbar, configs);
});
