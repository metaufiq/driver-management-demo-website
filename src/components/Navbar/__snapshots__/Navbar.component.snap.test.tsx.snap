// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarSnapshot should render normal Navbar with visible value is equal to false 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi GyWKw"
  >
    <a
      class="sc-eCYdqJ kQaECE"
      href="#"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </a>
    <div
      class="sc-jSMfEi kJYybr"
    >
      <p
        class="sc-dkzDqf hMVRxV"
      >
        Hello,
      </p>
      <p
        class="sc-dkzDqf sc-hKMtZM hMVRxV idCBRc"
      >
        Shipper User
      </p>
      <img
        class="sc-bczRLJ hGUeOa"
        height="30%"
        src="https://randomuser.me/api/portraits/med/men/18.jpg"
        width="30%"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`NavbarSnapshot should render normal Navbar with visible value is equal to true 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi gCnGDm"
  >
    <a
      class="sc-eCYdqJ hAAPjp"
      href="#"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </a>
    <div
      class="sc-jSMfEi kJYybr"
    >
      <p
        class="sc-dkzDqf hMVRxV"
      >
        Hello,
      </p>
      <p
        class="sc-dkzDqf sc-hKMtZM hMVRxV idCBRc"
      >
        Shipper User
      </p>
      <img
        class="sc-bczRLJ hGUeOa"
        height="30%"
        src="https://randomuser.me/api/portraits/med/men/18.jpg"
        width="30%"
      />
    </div>
  </div>
</DocumentFragment>
`;
